docs/openshift.asciidoc

```diff
@@ -0,0 +1,140 @@
+[id="{p}-openshift"]
+== Openshift
+
+This section will help you run ECK on Openshift.
+
+NOTE: Only Elasticsearch and Kibana are supported on Openshift.
+
+=== Prerequisites 
+
+. The following instructions must be run as the `system:admin` user or as a `User` with some privileges to create  `Projects`, `CRDs` and `RBACs` resources at the cluster level.
+
+. Set virtual memory settings on the nodes
++
+Before deploying an Elasticsearch cluster with ECK you must ensure that the `vm.max_map_count` setting is correctly applied on all the nodes of your cluster. This is because the Pods created by ECK are likely to run with the `restricted` SCC, they will run with a limited set of privileges and will not be able to tune this setting. You can refer to the following https://www.elastic.co/guide/en/elasticsearch/reference/current/vm-max-map-count.html[documentation] for more information.
+
+=== Deploy the operator
+. Apply the all in one template as described in the quickstart
++
+[source,shell]
+----
+oc apply -f https://download.elastic.co/downloads/eck/0.9.0/all-in-one.yaml
+----
+
+. [Optional] If the Software Defined Network is configured with the `ovs-multitenant` plug-in you need to allow the `elastic-system` namespace to access other pods and services in the cluster: 
++
+[source,shell]
+----
+oc adm pod-network make-projects-global elastic-system
+----
+
+. Create a namespace that will hold the Elastic resources (Elasticsearch, Kibana):
++
+[source,shell]
+----
+oc new-project elastic # creates the elastic project
+----
++
+By default the operator will watch and create resources in the `default` namespace. You need to patch the operator to manage resources in an other namespace.
```

Comment:
Might be worth providing an example of listening cluster wide here as well, since I imagine that is a common use case

Reply:
I would be tempted to say that it is not specific to Openshift, so it should be documented elsewhere. That said it's an interesting point, I'm not sure we have a document explaining how to do that.

Reply:
Yeah that's a good point. We should definitely add that to a larger configuration doc and perhaps link to it from here